src/xenia/base/testing/memory_test.cc

```diff
@@ -414,6 +417,58 @@ TEST_CASE("copy_and_swap_16_in_32_unaligned", "Copy and Swap") {
   REQUIRE(true == true);
 }
 
+TEST_CASE("create_and_close_file_mapping", "Virtual Memory Mapping") {
+  auto path = fmt::format("xenia_test_{}", Clock::QueryHostTickCount());
+  auto memory = xe::memory::CreateFileMappingHandle(
+      path, 0x100, xe::memory::PageAccess::kReadWrite, false);
+  REQUIRE(memory);
+  xe::memory::CloseFileMappingHandle(memory, path);
+}
+
+TEST_CASE("map_view", "Virtual Memory Mapping") {
+  auto path = fmt::format("xenia_test_{}", Clock::QueryHostTickCount());
+  const size_t length = 0x100;
+  auto memory = xe::memory::CreateFileMappingHandle(
+      path, length, xe::memory::PageAccess::kReadWrite, false);
+  REQUIRE(memory);
+
+  uintptr_t address = 0x100000000;
+  auto view =
+      xe::memory::MapFileView(memory, reinterpret_cast<void*>(address), length,
+                              xe::memory::PageAccess::kReadWrite, 0);
+  REQUIRE(reinterpret_cast<uintptr_t>(view) == address);
+
+  xe::memory::UnmapFileView(memory, reinterpret_cast<void*>(address), length);
+  xe::memory::CloseFileMappingHandle(memory, path);
+}
+
+TEST_CASE("read_write_view", "Virtual Memory Mapping") {
+  const size_t length = 0x100;
+  auto path = fmt::format("xenia_test_{}", Clock::QueryHostTickCount());
+  auto memory = xe::memory::CreateFileMappingHandle(
+      path, length, xe::memory::PageAccess::kReadWrite, false);
+  REQUIRE(memory);
+
+  uintptr_t address = 0x100000000;
+  auto view =
+      xe::memory::MapFileView(memory, reinterpret_cast<void*>(address), length,
+                              xe::memory::PageAccess::kReadWrite, 0);
+  REQUIRE(reinterpret_cast<uintptr_t>(view) == address);
+
+  for (uint32_t i = 0; i < length; i += sizeof(uint8_t)) {
+    auto p_value = reinterpret_cast<uint8_t*>(address + i);
+    *p_value = i;
```

Comment:
`for (uint32_t i = 0; i < length; i += sizeof(uint8_t))` - this two loops lead to test fail.
`*p_value = i;` - this line lead to fail: https://ci.appveyor.com/project/benvanik/xenia/builds/32094475/job/1v3ojal5dj5g66bv#L1197
```
C:\projects\xenia\src\xenia\base\testing\memory_test.cc(445): FAILED:
  {Unknown expression after the reported line}
with expansion:
due to a fatal error condition:
  SIGSEGV - Segmentation violation signal
```
Can you check it on Windows?

Reply:
I can replicate. It seems that using `CreateFileMappingHandle` with `commit=true` doesn't segfault.